core/Cargo.toml

```diff
@@ -0,0 +1,17 @@
+[package]
+name = "jf-primitives-core"
```

Comment:
Could this just be `jf-core` or maybe merged with `jf-rescue`?

Reply:
Oh well, we prolly can't merge with `jf-rescue`; I see we use `VerificationResult` in many crates.

Reply:
@alxiong is suggesting moving `VerificationResult` to `jf-utils`: https://github.com/EspressoSystems/jellyfish/pull/556#pullrequestreview-2011503266

However, I'm hesitant of doing so, because it adds one more dependency for this crate.
If you have more ideas.

Reply:
it comes down to the role of `VerificationResult` (a type alias to nudge for better error handling, without any extra fancy features/methods), this is more like a type hint, or a `#[must_use]` declaration, IMO.

giving it a second thought, i actually tend to think that we shouldn't export this, but only declare it locally as a type alias and remove this dep completely. 

for example this works:

```rust
mod core {
    // therefore this can be completely removed, 
    // and only declared at `mod vdf` level as a crate-level type alias
    pub type VerificationResult = Result<(), ()>;
}

mod vdf {
    use super::core::*;
    pub trait VDF {
        fn eval() -> Result<VerificationResult, ()>;
    }
}

mod vdf_impl {
    use super::vdf::VDF;
    struct VDF1;
    impl VDF for VDF1 {
        // I don't need to import `VerificationResult` to compile 
        fn eval() -> Result<Result<(), ()>, ()> {
            Err(())
        }
    }
}
```

Reply:
Originally in the comments it says
```rust
/// Intended as the return type for verification of proofs, signatures, etc.
/// Recommended for use in the nested [`Result`] pattern: see <https://sled.rs/errors>.
```

If we are not exporting this, are we still propagating this pattern?
For example, the merkle tree is refactored to use the verification result. Should we change it back or move the trait definition to `jf-traits`?

Reply:
I missed this comment until now. Not sure if still relevant but: I'm happy to stop propagating this pattern.